FoodFite/Models/UserProfile.cs

```diff
@@ -31,36 +31,41 @@ public class UserProfile
         }
 
         public List<Item> ListProtection(){
-            List<Item> list = Inventory.Where(x => x.Throwable == false).ToList<Item>();
-            list.Add(Clothes);
-            return list;
+            return Inventory.Where(x => x.Throwable == false).ToList<Item>();
         }
 
         public void ChangeClothes(Protection newClothes){
-            Protection oldClothes = Clothes;
-            Inventory.Remove(newClothes);
-            Clothes = newClothes;
-            if(oldClothes != null){
-                Inventory.Add(oldClothes);
+            if(newClothes != null){
+                if (Clothes != null){
+                    int index = Inventory.FindIndex(item => item.Name == Clothes.Name);
+                    Inventory.RemoveAt(index);
+                }
+                Clothes = newClothes;
+                Inventory.Add(newClothes);
             }
         }
 
         public double ThrowFood(Food food, UserProfile target){
             double damage = food.Attack(0);
 
             if(!food.hasAmmo()){
-                Inventory.Remove(food);
+                int index = Inventory.FindIndex(item => item.Name == food.Name);
+                Inventory.RemoveAt(index);
```

Comment:
if you remove it from the inventory array, you also need to remove it from the foodmap.

Reply:
fixed